xbmc/music/windows/GUIWindowVisualisation.cpp

```diff
@@ -240,3 +241,32 @@ void CGUIWindowVisualisation::FrameMove()
   }
   CGUIWindow::FrameMove();
 }
+
+void CGUIWindowVisualisation::ToggleOSD()
+{
+  CGUIDialogMusicOSD *pOSD = static_cast<CGUIDialogMusicOSD *>(g_windowManager.GetWindow(WINDOW_DIALOG_MUSIC_OSD));
+  if (pOSD)
+  {
+    if (pOSD->IsDialogRunning())
+    {
+      pOSD->Close();
+    }
+    else
```

Comment:
I'd rewrite this block too:
```C++
    else if (g_infoManager.GetShowInfo() &&
          CSettings::Get().GetBool("pvrplayback.confirmchannelswitch") &&
          g_application.CurrentFileItem().HasChannel())
    {
      // If only channel info but no player controls are currently visible
      // and setting "confirm channel switch with OK" is active, do a channel
      // switch instead of turning on the the player controls.
      g_application.m_pPlayer->SwitchChannel(g_application.CurrentFileItem().GetPVRChannelInfoTag());
      return;
    }
```